Content.Client/Administration/UI/CustomControls/ConfirmWindow.xaml.cs

```diff
@@ -0,0 +1,36 @@
+using Content.Client.Stylesheets;
+using System.Numerics;
+using Robust.Client.AutoGenerated;
+using Robust.Client.Console;
+using Robust.Client.UserInterface.CustomControls;
+using Robust.Client.UserInterface.XAML;
+
+namespace Content.Client.Administration.UI.CustomControls;
+
+[GenerateTypedNameReferences]
+public sealed partial class ConfirmWindow : DefaultWindow
+{
+    public event Action? OnDisposed;
+    public string? Command { get; }
+
+    public ConfirmWindow(string command)
+    {
+        this.Command = command;
+        RobustXamlLoader.Load(this);
+        IoCManager.InjectDependencies(this);
+
+        Title = Loc.GetString("confirm-command-title", ("command", this.Command));
+
+        ConfirmButton.AddStyleClass(StyleNano.StyleClassButtonColorRed);
+
+        CancelButton.OnPressed += _ =>
+        {
+            Close();
+        };
+        ConfirmButton.OnPressed += _ =>
+        {
+            IoCManager.Resolve<IClientConsoleHost>().ExecuteCommand(this.Command);
```

Comment:
This very heavily ties the ConfirmWindow to the usage in the admin window and duplicates the ExecuteCommand() action. It would be better to make this window more generic so it can be used in different places.

Suggest that ConfirmWindow exposes a OnCancel/OnConfirm action, which you can subscribe to from the CommandButton and remove this duplication.